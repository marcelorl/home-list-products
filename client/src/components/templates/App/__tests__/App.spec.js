import createTestContext from 'react-cosmos-test/enzyme'

import fixture from '../App.fixture'

const { mount, getWrapper } = createTestContext({ fixture })

beforeEach(mount)

describe('#App template', () => {
  it('renders component', () => {
    expect(getWrapper()).toMatchSnapshot()
  })

  describe('#breadcrumb', () => {
    it('should exist', () => {
      expect(getWrapper('.container ul.breadcrumb')).toHaveLength(1)
    })

    it('should have 3 items', () => {
      expect(getWrapper('.container ul.breadcrumb li')).toHaveLength(3)
    })
  })

})
