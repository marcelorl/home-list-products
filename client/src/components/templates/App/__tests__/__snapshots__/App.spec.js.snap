// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#App template renders component 1`] = `
<App
  products={
    Object {
      "error": "",
      "list": Array [
        Object {
          "brandName": "brandName",
          "description": "this is just a short description",
          "id": 1,
          "image": "test.png",
          "price": 1000,
          "priceFormatted": "1000 R$",
          "slug": "http://test.com",
        },
        Object {
          "brandName": "brandName",
          "description": "this is just a short description",
          "id": 2,
          "image": "test.png",
          "price": 1000,
          "priceFormatted": "1000 R$",
          "slug": "http://test.com",
        },
        Object {
          "brandName": "brandName",
          "description": "this is just a short description",
          "id": 3,
          "image": "test.png",
          "price": 1000,
          "priceFormatted": "1000 R$",
          "slug": "http://test.com",
        },
      ],
      "loading": false,
    }
  }
>
  <styled.div>
    <div
      className="sc-gZMcBi cQTtLh"
    >
      <Header>
        <Styled(Navbar)
          className="d-flex justify-content-center container-fluid px-0"
          color="faded"
          expand="lg"
          light={true}
        >
          <Navbar
            className="d-flex justify-content-center container-fluid px-0 sc-bdVaJa iQTWxT"
            color="faded"
            expand="lg"
            light={true}
            tag="nav"
          >
            <nav
              className="d-flex justify-content-center container-fluid px-0 sc-bdVaJa iQTWxT navbar navbar-expand-lg navbar-light bg-faded"
            >
              <styled.img
                alt="logo"
                src="logo.png"
              >
                <img
                  alt="logo"
                  className="sc-bwzfXH kXPULw"
                  src="logo.png"
                />
              </styled.img>
            </nav>
          </Navbar>
        </Styled(Navbar)>
      </Header>
      <SubHeader>
        <styled.div>
          <div
            className="sc-ifAKCX iVGUfa"
          >
            <Container
              tag="div"
            >
              <div
                className="container"
              >
                <Breadcrumb
                  items={
                    Array [
                      Object {
                        "label": "Início",
                        "link": "/inicio",
                      },
                      Object {
                        "label": "Homem",
                        "link": "/homem",
                      },
                      Object {
                        "label": "Novidades Internacionais",
                        "link": "/",
                      },
                    ]
                  }
                >
                  <styled.ul
                    className="breadcrumb"
                  >
                    <ul
                      className="breadcrumb sc-htpNat hGSyn"
                    >
                      <styled.li
                        key="Início"
                      >
                        <li
                          className="sc-bxivhb QZhb"
                        >
                          <a
                            href="/inicio"
                          >
                            Início
                          </a>
                        </li>
                      </styled.li>
                      <styled.li
                        key="Homem"
                      >
                        <li
                          className="sc-bxivhb QZhb"
                        >
                          <a
                            href="/homem"
                          >
                            Homem
                          </a>
                        </li>
                      </styled.li>
                      <styled.li
                        key="Novidades Internacionais"
                      >
                        <li
                          className="sc-bxivhb QZhb"
                        >
                          <a
                            href="/"
                          >
                            Novidades Internacionais
                          </a>
                        </li>
                      </styled.li>
                    </ul>
                  </styled.ul>
                </Breadcrumb>
                <div>
                  <h1>
                    NOVIDADES DA MODA MASCULINA INTERNACIONAL
                  </h1>
                </div>
                <div
                  className="subheader__description"
                >
                  O melhor da moda masculina atual e as peças mais recentes das marcas internacionais chegaram à seleção online da Farfetch. Descubra novidades da moda masculina e tendências que grandes designers para renovar o guarda-roupas, de 
                  <a
                    href="/"
                  >
                    Salvatore Ferragamo
                  </a>
                   à 
                  <a
                    href="/"
                  >
                    Burberry
                  </a>
                  .
                </div>
              </div>
            </Container>
          </div>
        </styled.div>
      </SubHeader>
      <ListProducts
        products={
          Object {
            "error": "",
            "list": Array [
              Object {
                "brandName": "brandName",
                "description": "this is just a short description",
                "id": 1,
                "image": "test.png",
                "price": 1000,
                "priceFormatted": "1000 R$",
                "slug": "http://test.com",
              },
              Object {
                "brandName": "brandName",
                "description": "this is just a short description",
                "id": 2,
                "image": "test.png",
                "price": 1000,
                "priceFormatted": "1000 R$",
                "slug": "http://test.com",
              },
              Object {
                "brandName": "brandName",
                "description": "this is just a short description",
                "id": 3,
                "image": "test.png",
                "price": 1000,
                "priceFormatted": "1000 R$",
                "slug": "http://test.com",
              },
            ],
            "loading": false,
          }
        }
      >
        <styled.div>
          <div
            className="sc-dnqmqq irsWwO"
          >
            <Container
              tag="div"
            >
              <div
                className="container"
              >
                <styled.div>
                  <div
                    className="sc-iwsKbI Jjtqy"
                  >
                    <Item
                      brandName="brandName"
                      description="this is just a short description"
                      id={1}
                      image="test.png"
                      key="1"
                      price={1000}
                      priceFormatted="1000 R$"
                      slug="http://test.com"
                    >
                      <styled.div
                        className="item__container"
                      >
                        <div
                          className="item__container sc-EHOje gUdhUV"
                        >
                          <Card
                            tag="div"
                          >
                            <div
                              className="card"
                            >
                              <Styled(CardImg)
                                alt="http://test.com"
                                src="test.png"
                                top={true}
                                width="100%"
                              >
                                <CardImg
                                  alt="http://test.com"
                                  className="sc-bZQynM iOWXCy"
                                  src="test.png"
                                  tag="img"
                                  top={true}
                                  width="100%"
                                >
                                  <img
                                    alt="http://test.com"
                                    className="sc-bZQynM iOWXCy card-img-top"
                                    src="test.png"
                                    width="100%"
                                  />
                                </CardImg>
                              </Styled(CardImg)>
                              <CardBody
                                tag="div"
                              >
                                <div
                                  className="card-body"
                                >
                                  <Styled(CardTitle)>
                                    <CardTitle
                                      className="sc-gzVnrw dvnGhf"
                                      tag="h5"
                                    >
                                      <h5
                                        className="sc-gzVnrw dvnGhf card-title"
                                      >
                                        brandName
                                      </h5>
                                    </CardTitle>
                                  </Styled(CardTitle)>
                                  <Styled(CardSubtitle)>
                                    <CardSubtitle
                                      className="sc-htoDjs fNBXrb"
                                      tag="h6"
                                    >
                                      <h6
                                        className="sc-htoDjs fNBXrb card-subtitle"
                                      >
                                        this is just a short description
                                      </h6>
                                    </CardSubtitle>
                                  </Styled(CardSubtitle)>
                                  <div>
                                    <div
                                      className="item__price"
                                    >
                                      1000 R$
                                    </div>
                                    <div
                                      className="item__parcels"
                                    >
                                      12x 83.33 €
                                    </div>
                                  </div>
                                </div>
                              </CardBody>
                            </div>
                          </Card>
                        </div>
                      </styled.div>
                    </Item>
                    <Item
                      brandName="brandName"
                      description="this is just a short description"
                      id={2}
                      image="test.png"
                      key="2"
                      price={1000}
                      priceFormatted="1000 R$"
                      slug="http://test.com"
                    >
                      <styled.div
                        className="item__container"
                      >
                        <div
                          className="item__container sc-EHOje gUdhUV"
                        >
                          <Card
                            tag="div"
                          >
                            <div
                              className="card"
                            >
                              <Styled(CardImg)
                                alt="http://test.com"
                                src="test.png"
                                top={true}
                                width="100%"
                              >
                                <CardImg
                                  alt="http://test.com"
                                  className="sc-bZQynM iOWXCy"
                                  src="test.png"
                                  tag="img"
                                  top={true}
                                  width="100%"
                                >
                                  <img
                                    alt="http://test.com"
                                    className="sc-bZQynM iOWXCy card-img-top"
                                    src="test.png"
                                    width="100%"
                                  />
                                </CardImg>
                              </Styled(CardImg)>
                              <CardBody
                                tag="div"
                              >
                                <div
                                  className="card-body"
                                >
                                  <Styled(CardTitle)>
                                    <CardTitle
                                      className="sc-gzVnrw dvnGhf"
                                      tag="h5"
                                    >
                                      <h5
                                        className="sc-gzVnrw dvnGhf card-title"
                                      >
                                        brandName
                                      </h5>
                                    </CardTitle>
                                  </Styled(CardTitle)>
                                  <Styled(CardSubtitle)>
                                    <CardSubtitle
                                      className="sc-htoDjs fNBXrb"
                                      tag="h6"
                                    >
                                      <h6
                                        className="sc-htoDjs fNBXrb card-subtitle"
                                      >
                                        this is just a short description
                                      </h6>
                                    </CardSubtitle>
                                  </Styled(CardSubtitle)>
                                  <div>
                                    <div
                                      className="item__price"
                                    >
                                      1000 R$
                                    </div>
                                    <div
                                      className="item__parcels"
                                    >
                                      12x 83.33 €
                                    </div>
                                  </div>
                                </div>
                              </CardBody>
                            </div>
                          </Card>
                        </div>
                      </styled.div>
                    </Item>
                    <Item
                      brandName="brandName"
                      description="this is just a short description"
                      id={3}
                      image="test.png"
                      key="3"
                      price={1000}
                      priceFormatted="1000 R$"
                      slug="http://test.com"
                    >
                      <styled.div
                        className="item__container"
                      >
                        <div
                          className="item__container sc-EHOje gUdhUV"
                        >
                          <Card
                            tag="div"
                          >
                            <div
                              className="card"
                            >
                              <Styled(CardImg)
                                alt="http://test.com"
                                src="test.png"
                                top={true}
                                width="100%"
                              >
                                <CardImg
                                  alt="http://test.com"
                                  className="sc-bZQynM iOWXCy"
                                  src="test.png"
                                  tag="img"
                                  top={true}
                                  width="100%"
                                >
                                  <img
                                    alt="http://test.com"
                                    className="sc-bZQynM iOWXCy card-img-top"
                                    src="test.png"
                                    width="100%"
                                  />
                                </CardImg>
                              </Styled(CardImg)>
                              <CardBody
                                tag="div"
                              >
                                <div
                                  className="card-body"
                                >
                                  <Styled(CardTitle)>
                                    <CardTitle
                                      className="sc-gzVnrw dvnGhf"
                                      tag="h5"
                                    >
                                      <h5
                                        className="sc-gzVnrw dvnGhf card-title"
                                      >
                                        brandName
                                      </h5>
                                    </CardTitle>
                                  </Styled(CardTitle)>
                                  <Styled(CardSubtitle)>
                                    <CardSubtitle
                                      className="sc-htoDjs fNBXrb"
                                      tag="h6"
                                    >
                                      <h6
                                        className="sc-htoDjs fNBXrb card-subtitle"
                                      >
                                        this is just a short description
                                      </h6>
                                    </CardSubtitle>
                                  </Styled(CardSubtitle)>
                                  <div>
                                    <div
                                      className="item__price"
                                    >
                                      1000 R$
                                    </div>
                                    <div
                                      className="item__parcels"
                                    >
                                      12x 83.33 €
                                    </div>
                                  </div>
                                </div>
                              </CardBody>
                            </div>
                          </Card>
                        </div>
                      </styled.div>
                    </Item>
                  </div>
                </styled.div>
              </div>
            </Container>
          </div>
        </styled.div>
      </ListProducts>
    </div>
  </styled.div>
</App>
`;
